eclair-core/src/main/scala/fr/acinq/eclair/crypto/Mac.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.crypto
+
+import fr.acinq.bitcoin.ByteVector32
+import org.spongycastle.crypto.digests.SHA256Digest
+import org.spongycastle.crypto.macs.HMac
+import org.spongycastle.crypto.params.KeyParameter
+import scodec.bits.ByteVector
+
+/**
+  * Created by t-bast on 04/07/19.
+  */
+
+/**
+  * Create and verify message authentication codes.
+  */
+trait Mac {
+
+  def mac(message: ByteVector): ByteVector32
+
+  def verify(mac: ByteVector32, message: ByteVector): Boolean
+
+}
+
+case class Hmac256(key: ByteVector) extends Mac {
+
+  override def mac(message: ByteVector): ByteVector32 = Mac.hmac256(key, message)
+
+  override def verify(mac: ByteVector32, message: ByteVector): Boolean = this.mac(message) === mac
```

Comment:
Looking at scodec javadoc, triple equals seems to be the correct choice, but we are using double equals everywhere else and never had issues. Not sure what's the best practice here.